pkg/container-collection/container-collection.go

```diff
@@ -422,11 +425,22 @@ func (cc *ContainerCollection) Close() {
 		panic("ContainerCollection is not initialized or has been closed")
 	}
 
+	// TODO: it's not clear if we want/can allow to re-initialize
+	// this instance yet, so we don't set cc.initialized = false.
+	cc.closed = true
+
 	for _, f := range cc.cleanUpFuncs {
 		f()
 	}
 
-	// TODO: it's not clear if we want/can allow to re-initialize
-	// this instance yet, so we don't set cc.initialized = false.
-	cc.closed = true
+	// Similar to RemoveContainer() on all containers but without publishing
+	// events.
+	cc.containers.Range(func(key, value interface{}) bool {
+		c := value.(*Container)
+		for _, f := range c.cleanUpFuncs {
+			f()
+		}
+		cc.containers.Delete(c)
+		return true
+	})
```

Comment:
The callback from SetFinalizer is called by the garbage collector and it might take many seconds before it happens. I think fd should be closed as soon as possible because:
1. In a scenario where containers are started and stopped quickly, ig might keep many file descriptor without needing it.
2. We have a test that checks for file descriptor leaks after the ContainerCollection is closed, and it fails if the mntns fd is not closed fast enough.
```
pkg/ig-manager/ig-manager_test.go:129:func TestClose(t *testing.T)
```